lanelet2/lanelet2_core/src/LaneletMap.cpp

```diff
@@ -774,6 +774,44 @@ void LaneletMap::add(Area area) {
   }
 }
 
+void LaneletMap::remove(Lanelet ll, const RegulatoryElementPtr& regElem)
+{
+  if (ll.id() == InvalId) {
+    throw InvalidInputError("Lanelet element with InvalId is passed to update()!");
+  }
+  else if (!laneletLayer.exists(ll.id()))
+  {
+    throw InvalidInputError("Lanelet element that is not in the map is passed to update()!");
+  }
+  if (!regElem) {
+    throw NullptrError("Empty regulatory element passed to update()!");
+  }
+  
+  if (regElem->id() == InvalId) {
+    throw InvalidInputError("Regulatory element with InvalId is passed to update()!");
+  }
+  else if (!regulatoryElementLayer.exists(regElem->id())) {
+    throw InvalidInputError("Id of the regulatory element is not registered in the map");
+  }
+ 
+  lanelet::Lanelets parent_llts = laneletLayer.findUsages(regElem);
+
+  if (regulatoryElementLayer.exists(regElem->id()) && std::find(parent_llts.begin(), parent_llts.end(), ll) == parent_llts.end()) {
```

Comment:
The regulatoryElementLayer.exists(regElem->id()) check here is redundant given if statement above.